lib/transition/unpack.c

```diff
@@ -228,10 +498,31 @@ unpack_graph(xmlNode * xml_graph, const char *reference)
 
             if (new_synapse != NULL) {
                 new_graph->synapses = g_list_append(new_graph->synapses, new_synapse);
+                probe_actions = prepend_probe_action(probe_actions, new_synapse);
+                start_actions = prepend_start_action(start_actions, new_synapse);
```

Comment:
Injecting meta-data actions while unpacking the graph invalidates iterators of graph->synapses. I used this approach in the previous implementation and you pointed this out.